packages/lodestar/src/network/peers/utils/peerMapDelay.ts

```diff
@@ -0,0 +1,36 @@
+import PeerId from "peer-id";
+import {PeerMap} from "../../../util/peerMap";
+
+export class PeerMapDelay {
+  interval: number;
+  lastMsMap = new PeerMap<number>();
+  constructor(interval: number) {
+    this.interval = interval;
+  }
+
+  /** timeToReq = 0 -> Request as soon as pingAndStatusTimeouts() is called */
+  requestNow(peer: PeerId): void {
+    this.requestAfter(peer, -1);
+  }
+
+  /** timeToReq = now() -> Request after `INTERVAL` */
+  requestAfter(peer: PeerId, ms = this.interval): void {
+    this.lastMsMap.set(peer, Date.now() - this.interval + ms);
+  }
+
+  /** Return array of peers with expired interval + call requestAfter on them */
+  pollNext(): PeerId[] {
+    const peers: PeerId[] = [];
+    for (const [peer, lastMs] of this.lastMsMap.entries()) {
+      if (Date.now() - lastMs > this.interval) {
```

Comment:
I see your point but I'm not sure it would have any impact. This data structure is somewhat equivalent to discv5 timeout map, but with a single setInterval in the parent class. If there are small differences in the timing it can cause some peers to be poll'ed in the next interval run. If a peer is ping'ed or status'ed -+ 2 seconds shouldn't matter much.